pkg/client/typed/discovery/discovery_client.go

```diff
@@ -163,6 +171,50 @@ func (d *DiscoveryClient) ServerResources() (map[string]*unversioned.APIResource
 	return result, nil
 }
 
+// serverPreferredResources returns the supported resources with the version preferred by the
+// server. If namespaced is true, only namespaced resources will be returned.
+func (d *DiscoveryClient) serverPreferredResources(namespaced bool) ([]unversioned.GroupVersionResource, error) {
```

Comment:
API versions of third party resource data objects are unrelated to each other. If the GC controller is watching v1 of myresource.mygroup.io because v1 is the preferred version, it won't know about v2 objects of that group/resource.